ingestion/src/metadata/ingestion/source/database/trino.py

```diff
@@ -154,6 +160,48 @@ def create(cls, config_dict, metadata_config: OpenMetadataConnection):
             )
         return cls(config, metadata_config)
 
-    def get_database_names(self) -> Iterable[str]:
+    def set_inspector(self, database_name: str) -> None:
+        """
+        When sources override `get_database_names`, they will need
+        to setup multiple inspectors. They can use this function.
+        :param database_name: new database to set
+        """
+        logger.info(f"Ingesting from catalog: {database_name}")
+
+        new_service_connection = deepcopy(self.service_connection)
+        new_service_connection.catalog = database_name
+        self.engine = get_connection(new_service_connection)
         self.inspector = inspect(self.engine)
-        yield self.trino_connection.catalog
+
+    def get_database_names(self) -> Iterable[str]:
+        configured_catlog = self.trino_connection.catalog
+        if configured_catlog:
```

Comment:
nit - typo in `catlog`

Reply:
needs to be changed in both trino and presto

Reply:
Solved in both.🙏